### DISCUSSION TITLE
Tried to upgrade Teleport to version 12.1.1 and messed up.

### DISCUSSION THREAD

**Discussion by 420aakura** [gravitational/teleport#23106] [Q&A] [answered]
_English is not my first language so please ignore my language mistakes_

I tried installing and using teleport for SSH three days ago following [this article from documentation](https://goteleport.com/docs/try-out-teleport/linux-server/). It worked perfectly. But, today I got a banner on the web ui as new version is out. So, I thought updating the teleport will be just `sudo apt install --only-upgrade teleport` but, after the update, proxy website got down.

Trying to fix things, I thought package was not installed correctly and tried reinstalling from start. And package is installed correctly but now I am getting errors while:
`sudo tctl users add teleport-admin --roles=editor,access --logins=root,user1`

Output:
`[CLIENT]       Cannot connect to the auth server: failed direct dial to auth server: dial tcp 0.0.0.0:3025: connect: connection refused
	dial tcp 0.0.0.0:3025: connect: connection refused.
Is the auth server running on "0.0.0.0:3025"?`

Tried to see if 3025 port is listening or not:
`netstat -alnp | grep LISTEN`
which outputs:
```tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -                   
tcp6       0      0 :::22                   :::*                    LISTEN      -                   
unix  2      [ ACC ]     STREAM     LISTENING     102382   19549/systemd        /run/user/1008/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     102390   19549/systemd        /run/user/1008/bus
unix  2      [ ACC ]     STREAM     LISTENING     102392   19549/systemd        /run/user/1008/gnupg/S.dirmngr
unix  2      [ ACC ]     STREAM     LISTENING     102394   19549/systemd        /run/user/1008/gnupg/S.gpg-agent.browser
unix  2      [ ACC ]     STREAM     LISTENING     102396   19549/systemd        /run/user/1008/gnupg/S.gpg-agent.extra
unix  2      [ ACC ]     STREAM     LISTENING     102398   19549/systemd        /run/user/1008/gnupg/S.gpg-agent.ssh
unix  2      [ ACC ]     STREAM     LISTENING     102400   19549/systemd        /run/user/1008/gnupg/S.gpg-agent
unix  2      [ ACC ]     STREAM     LISTENING     10096    -                    /run/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     10098    -                    /run/systemd/userdb/io.systemd.DynamicUser
unix  2      [ ACC ]     STREAM     LISTENING     10099    -                    /run/systemd/io.system.ManagedOOM
unix  2      [ ACC ]     STREAM     LISTENING     10109    -                    /run/systemd/fsck.progress
unix  2      [ ACC ]     STREAM     LISTENING     10117    -                    /run/systemd/journal/stdout
unix  2      [ ACC ]     SEQPACKET  LISTENING     10119    -                    /run/udev/control
unix  2      [ ACC ]     STREAM     LISTENING     10144    -                    /run/systemd/journal/io.systemd.journal
unix  2      [ ACC ]     STREAM     LISTENING     10699    -                    /run/dbus/system_bus_socket
unix  2      [ ACC ]     STREAM     LISTENING     10703    -                    /run/docker.sock
unix  2      [ ACC ]     STREAM     LISTENING     10719    -                    /run/uuidd/request
unix  2      [ ACC ]     STREAM     LISTENING     11080    -                    /run/containerd/containerd.sock.ttrpc
unix  2      [ ACC ]     STREAM     LISTENING     11083    -                    /run/containerd/containerd.sock
unix  2      [ ACC ]     STREAM     LISTENING     11124    -                    /var/run/docker/metrics.sock
unix  2      [ ACC ]     STREAM     LISTENING     11273    -                    /var/run/docker/libnetwork/961376fc8390.soc
```

Tried [this issue #599](https://github.com/gravitational/teleport/issues/599). Still I am getting same output.

My teleport.yaml is:

```#
# A Sample Teleport configuration file.
#
# Things to update:
#  1. license.pem: You only need a license from https://dashboard.goteleport.com
#     if you are an Enterprise customer.
#
version: v3
teleport:
  nodename: boundary
  data_dir: /var/lib/teleport
  log:
    output: /var/lib/teleport/teleport.log 
    severity: INFO
    format:
      output: text
      extra_fields: [level, timestamp, component, caller]
  ca_pin: ""
  diag_addr: ""
auth_service:
  enabled: yes
  listen_addr: 0.0.0.0:3025
  cluster_name: teleport.xxxxxxxxx.com.xx
  proxy_listener_mode: multiplex
ssh_service:
  enabled: yes
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
proxy_service:
  https_cert_file: /var/lib/letsencrypt/live/xxxxxxxxxxx.com.xx/fullchain.pem
  https_key_file: /var/lib/letsencrypt/live/xxxxxxxxxxx.com.xx/privkey.pem
  enabled: yes
  web_listen_addr: 0.0.0.0:443
  public_addr: teleport.xxxxxxxxxxxx.com.xx:443
  ```
  
  I have no clue what to do now. Please help me understand the problem and how to solve it. Should I open a new issue?

**Comment by Valien** [comment]
TLS Routing is enabled by default (I see `multiplex` in your .yaml) and port `3025` is an API port for the nodes. Docs here - https://goteleport.com/docs/reference/networking/

Are you running your `proxy` node separate from the `auth` service? And when upgrading it is recommended to upgrade the `auth` service first.

Here are the docs for upgrading a cluster - https://goteleport.com/docs/management/operations/upgrading/

**Comment by 420aakura** [comment, marked as answer]
Solution for my case:
What happened was after installing from scratch, I wrote yaml again and after writing yaml, I ran certbot for certs but certs were not saved on path I mentioned in yaml file as in post. So, what I did is removed certs and created new and changed the file paths.

Then I ran:
```sudo teleport configure --test /etc/teleport.yaml``` // to test the yaml
```sudo systemctl restart teleport``` // restart teleport service
and the service is up again. But, I lost all the past configs and everything because I didn't know [this documentation exists](https://goteleport.com/docs/management/operations/upgrading/)
